src/CMakeLists.txt

```diff
@@ -87,6 +87,43 @@ target_compile_definitions(dlaf.prop
     $<$<BOOL:${DLAF_WITH_CUDA_MPI_RDMA}>:DLAF_WITH_CUDA_MPI_RDMA>
 )
 
+# Precompiled headers
+if(DLAF_WITH_PRECOMPILED_HEADERS)
+  add_library(dlaf.pch_lib OBJECT dummy.cpp)
+  target_link_libraries(dlaf.pch_lib PRIVATE dlaf.prop)
+  target_add_warnings(dlaf.pch_lib)
+
+  add_executable(dlaf.pch_exe dummy.cpp)
+  target_link_libraries(dlaf.pch_exe PRIVATE dlaf.prop)
+  target_add_warnings(dlaf.pch_exe)
+
+  set(precompiled_headers
+    <mpi.h>
+
+    $<$<BOOL:${DLAF_WITH_CUDA}>:<pika/cuda.hpp$<ANGLE-R>>
+    <pika/execution.hpp>
+    <pika/future.hpp>
+    <pika/mpi.hpp>
+    <pika/mutex.hpp>
+    <pika/program_options.hpp>
+    <pika/runtime.hpp>
+    <pika/thread.hpp>
+    <pika/unwrap.hpp>
+
+    <blas.hh>
+    # <lapack.hh>
```

Comment:
This is disabled because of the `I` definition that would get pulled in and break everything (gtest uses `I` as a template parameter). The header doesn't add much to compilation times so I've simply excluded it. The only workaround I can think of is to include a header last in this list which undefines `I`. What would you prefer? I can add a comment if we decide to keep it excluded.

Reply:
I think we can keep it excluded.

Reply:
I've added a comment explaining this. Please have a look and close if you're happy with it.